File: public/PublicMedia/static/solutions/nim/problem3.nim
# Purpose: Find the largest prime factor of the number 600851475143
# Author: Eben Quenneville 
# Date: 2023-01-29
proc problem3(): int64 =
    var
        num: int64 = 600_851_475_143
        i: int64 = 2
    while i*i <= num:
        if num mod i == 0:
            num = num div i
        else:
            i = i + 1
    return num

proc main() =
    let num = problem3()
    assert(num == 6857)
    echo("Output: ", num)

main()

File: public/PublicMedia/static/solutions/c/problem3.c
/*
Purpose: Find the largest prime factor of the number 600851475143

Author: Eben Quenneville
Date: 2022-09-05, updated 2023-01-27
*/

#include <assert.h>
#include <stdio.h>

long problem3() {
  long num = 600851475143;
  int i = 2;
  while (i * i <= num) {
    if (num % i == 0) {
      num /= i;
    } else {
      i += 1;
    }
  }
  return num;
}

int main() {
  long num = problem3();
  assert(num == 6857);
  printf("Output: %ld\n", num);
  return 0;
}
